proxy/middleware/wsauth.go

```diff
@@ -10,26 +10,20 @@ import (
 
 var protocolHeader = textproto.CanonicalMIMEHeaderKey("Sec-WebSocket-Protocol")
 
-const authHeaderKey = "Authorization"
-
 // This function does not verify the auth token. It just makes sure that the token is injected into Auth header
 func WsAuthMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if strings.HasPrefix(r.RequestURI, "/api/terminal/") {
 			protocolHeaderVal, ok := r.Header[protocolHeader]
 			if !ok || len(protocolHeaderVal) != 1 {
```

Comment:
When the token is missing, we'd send protocols as `['flightctl.standalone.auth', '']` . This raises an error when the WS is created, and that leads to an uncaught error. (the flow never gets here)

I think when there is no token we should only send the first protocol. That allows the flow to continue, then we get an error 401 which we capture correctly.

Reply:
thanks for catching that. I've updated the code to not send an empty string if there is no auth token.